### DISCUSSION TITLE
Suricata alerts appearing for a PASS rule

### DISCUSSION THREAD

**Discussion by branchnetconsulting** [Security-Onion-Solutions/securityonion#9036] [Unsupported Versions] [answered]
On the very latest SO release, this Suricata pass rule matches real traffic but instead of silently moving on, SO generates an alert about it.  It see it happening at multiple client sites, and don't know what to make of it.

pass tcp $HOME_NET [443,465,993,995] -> [172.16.0.0/12,192.168.0.0/16,10.0.0.0/8] any (msg:"skip SSL packets that don't involve Internet"; sid:500400; rev:1;)

Isn't it intended behavior for pass rules to never result in a Suricata alert being indexed?  Should such alerts even be appearing in the current eve...json file in the first place?

Kevin

**Comment by dougburks** [comment]
Is it generating an actual alert or is it just generating network metadata?

**Comment by branchnetconsulting** [comment]
They are showing up as alerts in the Alert->Suricata SO dashboard with this query criteria:
`event.dataset:alert AND event.module:suricata`

**Comment by dougburks** [comment]
Have you made any other changes to Suricata configuration like `action-order`?
https://suricata.readthedocs.io/en/suricata-6.0.0/configuration/suricata-yaml.html#action-order

**Comment by branchnetconsulting** [comment]
No, my suricata.yaml files are just the stock ones generated by SO.

**Comment by dougburks** [comment]
Are you able to share screenshots of the alert and its fields?

What is the output of the following?
```
sudo grep 500400 /opt/so/rules/nids/*.rules
```

**Comment by branchnetconsulting** [comment]
Sure

![image](https://user-images.githubusercontent.com/13170654/199730622-7713221b-b4ef-4755-87e1-c07e241119d9.png)

The _index field from above was
NAME-nsm-store01_searchnode:so-ids-2022.11.03

![image](https://user-images.githubusercontent.com/13170654/199731127-f2d68447-b277-4dfc-9795-26c5e3c11474.png)

```
# sudo grep 500400 /opt/so/rules/nids/*.rules
/opt/so/rules/nids/all.rules:alert dns $HOME_NET any -> any any (msg:"ETPRO MALWARE Observed DNS Query for Ursnif Domain"; dns.query; content:"500400.bit"; nocase; endswith; reference:md5,1d5608d71a70e889a6e62217537eb42d; classtype:trojan-activity; sid:2834077; rev:3; metadata:affected_product Windows_XP_Vista_7_8_10_Server_32_64_Bit, attack_target Client_Endpoint, created_at 2018_12_21, deployment Perimeter, former_category TROJAN, malware_family ursnif, performance_impact Low, signature_severity Major, updated_at 2020_09_16;)
/opt/so/rules/nids/all.rules:pass tcp $HOME_NET [443,465,993,995] -> [172.16.0.0/12,192.168.0.0/16,10.0.0.0/8] any (msg:"skip SSL packets that don't involve Internet"; sid:500400; rev:1;)
/opt/so/rules/nids/local.rules:pass tcp $HOME_NET [443,465,993,995] -> [172.16.0.0/12,192.168.0.0/16,10.0.0.0/8] any (msg:"skip SSL packets that don't involve Internet"; sid:500400; rev:1;)
```

Thanks!
Kevin

**Comment by dougburks** [comment]
I placed your rule on a couple of deployments and created traffic that matched the rule criteria, but Suricata did not generate any alerts. (Zeek did log the connection verifying that the sensor did indeed see the traffic.)

Have you checked your Suricata log for additional clues?

Have you tried restarting Suricata?

Have you tried rebooting the sensor?

Have you tried a fresh installation with no customization?

**Comment by branchnetconsulting** [comment, marked as answer]
I appreciate you taking time to test the scenario, and am baffled why I'm seeing this on multiple couple of SO stacks but it does not show up for you.  

I did check suricata.log and there is nothing that appears relevant to this issue appearing there.  

I tried a total reboot of manager and sensor but it made no difference.

Would you concur that a reasonable way to get rid of these unwanted "pass" alerts would be to have a localized copy of the suricata.alert pipeline with an added drop processor like below?

/opt/so/saltstack/local/salt/elasticsearch/files/ingest/suricata.alert

```
{
  "description" : "suricata.alert",
  "processors" : [
    { "rename":{ "field": "message2.alert",     "target_field": "rule", "ignore_failure": true  } },
    { "rename":{ "field": "rule.signature",     "target_field": "rule.name", "ignore_failure": true  } },
    { "rename":{ "field": "rule.ref",     "target_field": "rule.version", "ignore_failure": true  } },
    { "rename":{ "field": "rule.signature_id",     "target_field": "rule.uuid", "ignore_failure": true  } },
    { "rename":{ "field": "rule.signature_id",     "target_field": "rule.signature", "ignore_failure": true  } },
    { "rename":{ "field": "message2.payload_printable",     "target_field": "network.data.decoded", "ignore_failure": true  } },
    {
      "drop": {
        "if" : "ctx.rule?.rule =~ /^pass /"
      }
    },
    { "pipeline": { "name": "common.nids" } }
  ]
}

```

**Comment by branchnetconsulting** [comment]
That pipeline change appears to have resolved the problem for us, although the underlying mystery is unsolved.  Thanks for your help on this, Doug!

**Comment by branchnetconsulting** [comment]
I think this issue is a possible bug in Suricata itself, somehow triggered by Docker overlay network tunneled packets that match a pass rule.  I've opened an issue with them about it here:
https://forum.suricata.io/t/unexpectedly-seeing-alert-records-written-to-eve-json-when-pass-rules-are-triggered/2984
